R/output_analysis.R

```diff
@@ -12,5 +12,5 @@ growth_rate <- function(out) {
   days <- unique(out$day)
   pops <- sapply(days, function(x) sum(out$pop[out$day == x]))
   lambdas <- pops / c(NA, pops[1:(length(pops) - 1)])
-  exp(mean(log(lambdas), na.rm = TRUE))
+  exp(mean(log(lambdas[1:length(lambdas)-1]), na.rm = TRUE))
```

Comment:
Hmm, I'm mostly concerned that you and I are getting such different results, apparently from the same code -- I would expect them to be identical. 

I came across a difference before between the model results on an M1 MacBook and an Intel model. There was a difference in a floating point calculation that ended up having a surprisingly large effect. I wonder if it could be an issue like that. The results I shared were from an Intel mac, are you on M1?

I saved my version of the `results` object (a list of df's) to an `.Rdata` file. Emailing to that you b/c I can't upload it here. If it makes sense to you, we could compare that to the `results` you have, in order to isolate whether our versions of the model (i.e. `run()`) are producing different outputs.

Reply:
Oh wow, that is concerning that we get such different values just because of floating point calc. I am on M1. I will email you my results too so we can both poke around.

Reply:
Yikes, is there a way to deal with this by forcing a `round` at each step so it is always rounding to the same value?

Reply:
I'm not sure there's a way rounding would work reliably. Given that we're dealing with floating-point numbers, we might have to live with floating-point issues like this.

We could still think about the change to the growth rate function. You mentioned the issue being that the population drops off rapidly on the last day. Is that an issue you'd expect we'd have across the board with other model runs? If it's specific to this case, you might be right that it's hard to have a universal growth rate function that always works.

Perhaps worth discussing both these points next time we meet?

Reply:
yeah we should talk about this when we meet. I played around with it some and my values of `growth_rate` were super sensitive to the length of the time I ran the model. It depended when in the annual cycle it finished. I ended up making a new function `annual_growth_rate` that was less sensitive to the issue. I just pushed those changes. Would you mind looking at my new function?